src/rust/engine/options/src/args.rs

```diff
@@ -143,25 +145,74 @@ impl Args {
     }
 }
 
+pub(crate) struct ArgsTracker {
+    unconsumed_args: Mutex<HashSet<Arg>>,
```

Comment:
In case it's not obvious (although this is a common Rust idiom), `Arc` of (struct containing a) `Mutex` is what allows thread-safe interior mutability, unlike RefCell, which is not Sync.

Reply:
Because the ArgsReader is used (as a member of `sources: BTreeMap<Source, Arc<dyn OptionsSource>>`) in Python threads.